### PR TITLE
PP-4779: Add flags for Google/Apple Pay

### PR DESCRIPTION
**Pull request by oswaldquek**
We want the option of allowing services to turn Google/Apple Pay on/off
independently.

### CODE REVIEW

File: src/main/resources/migrations.xml [hunk 1]
```diff
@@ -957,4 +957,15 @@
         </addColumn>
     </changeSet>
 
+    <changeSet id="add allow_apple_pay and allow_google_pay to gateway_accounts table" author="">
```
Comment (kbottla) [inline src/main/resources/migrations.xml:960]: We need update scripts too or set defaults when adding new columns

File: src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountEntity.java [hunk 2]
```diff
@@ -110,6 +110,12 @@ public GatewayAccountEntity() {
     @Column(name = "allow_web_payments")
     private boolean allowWebPayments;
 
+    @Column(name = "allow_google_pay")
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountEntity.java]: Not exactly sure if this works as fields won't be available in environments without migration.

**Review by kbottla — CHANGES_REQUESTED** [review]

File: src/main/resources/migrations.xml [hunk 3]
```diff
@@ -957,4 +957,31 @@
...
+
+    <changeSet id="update allow_apple_pay and allow_google_pay with default values" author="">
+        <update tableName="gateway_accounts">
+            <column name="allow_apple_pay" value="false"/>
+            <where>allow_apple_pay IS NULL</where>
+        </update>
+        <update tableName="gateway_accounts">
+            <column name="allow_google_pay" value="false"/>
+            <where>allow_google_pay IS NULL</where>
+        </update>
```
Comment (alexbishop1) [inline src/main/resources/migrations.xml:985]: Probably safer to break this into two different changesets

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
One small change request

File: src/main/resources/migrations.xml (hunk 3 above)
Comment (oswaldquek) [reply src/main/resources/migrations.xml:985]: done

File: src/main/resources/migrations.xml [hunk 4]
```diff
@@ -957,4 +957,34 @@
...
+
+    <changeSet id="add default to allow_apple_pay and allow_google_pay" author="">
+        <addDefaultValue tableName="gateway_accounts" columnName="allow_apple_pay" defaultValue="false"/>
+        <addDefaultValue tableName="gateway_accounts" columnName="allow_google_pay" defaultValue="false"/>
+    </changeSet>
+
+    <changeSet id="update allow_apple_pay with default value" author="">
+        <update tableName="gateway_accounts">
+            <column name="allow_apple_pay" value="false"/>
+            <where>allow_apple_pay IS NULL</where>
```
Comment (kbottla) [inline src/main/resources/migrations.xml]: We don't need `IS NULL` condition as it is new column. Query is usually faster without additional clauses

File: src/main/resources/migrations.xml [hunk 5]
```diff
@@ -957,4 +957,34 @@
...
+        </update>
+    </changeSet>
+
+    <changeSet id="update allow_google_pay with default value" author="">
+        <update tableName="gateway_accounts">
+            <column name="allow_google_pay" value="false"/>
+            <where>allow_google_pay IS NULL</where>
```
Comment (kbottla) [inline src/main/resources/migrations.xml]: Same as above for `IS NULL`

**Review by alexbishop1 — APPROVED** [review]

**Review by kbottla — APPROVED** [review]

**Pull request merged**